src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -424,4 +512,14 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 		}
 		return result;
 	}
+
+	private _parseKeyValueAssignment(message: string): { key: string; value: string | undefined } {
+		const [key, ...rawValues] = message.split('=');
+		if (!rawValues.length) {
+			return { key, value: undefined }; // No '=' was found.
+		}
+		const rawValue = rawValues.join('=');
```

Comment:
Since we're handling this better should we instead use `.indexOf('=')` and `.slice()` to avoid the creation of extra array values?

Reply:
Of course, it's the more reasonable approach. Also, I was uneasy to leave this function untested, so moved it (along with the `deserializeMessage` which was depended on) out of the class and added some tests to verify its function.